packages/webapp/src/containers/Animals/Inventory/useAnimalOrBatchRemoval.ts

```diff
@@ -157,7 +162,35 @@ const useAnimalOrBatchRemoval = (
     }
   };
 
-  return { onConfirmRemoveAnimals, removalModalOpen, setRemovalModalOpen };
+  const getFinalizedTaskIdsSet = useCallback(() => {
+    return new Set([...completedTasks, ...abandonedTasks].map(({ task_id }) => task_id));
+  }, [completedTasks, abandonedTasks]);
+
+  const hasFinalizedTasks = useMemo(() => {
+    if (!removalModalOpen) {
+      return false;
+    }
+
+    const selectedInventoryIdsSet = new Set(selectedInventoryIds);
+    const finalizedTaskIdsSet = getFinalizedTaskIdsSet();
+
+    let inventoryFoundCount = 0;
+    for (let { id, tasks } of inventory) {
+      if (selectedInventoryIdsSet.has(id)) {
```

Comment:
Should we do this the other way round and iterate over the selected IDs and check if the corresponding entity exists in the inventory? The selected IDs would be a subset of the whole inventory so I think that should be faster than checking the whole inventory

Reply:
I initially considered iterating over `selectedInventoryIds` instead of `inventory` as suggested, and I came up with two approaches:

1. Format `inventory` into a Map:
    1. Convert `inventory` (`[{id, tasks}, {id, tasks}]`) to a map like `{[id]: tasks, [id]: tasks}`
    2. Then iterate over `selectedInventoryIds` to check if the corresponding tasks exist:
    ```
   for (let id of selectedInventoryIds) {
         if (inventoryMap[id].length) {
           return true;
         }
      }
   ```
2. Directly iterate `selectedInventoryIds` without formatting:
    ```
    for (let selectedInventoryId of selectedInventoryIds) {
       const { tasks } = inventory.find(({ id }) => id === selectedInventoryId);
       if (tasks.length) {
          return true;
       }
    }
    ```

With the first approach, we’d be iterating over `inventory`, so it might be more efficient to return the result early while iterating. The second approach felt a bit less efficient, though it should work fine for this feature.

Which option would you go with?

Reply:
Actually now that you described the two I think your implementation is probably the most performant one! I think option 2 presented here would be the simplest one / easier to read, but I'd keep this as is